.travis_docker.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+set -ev
+
+TARGET_IMAGE="${DOCKER_USERNAME}/${DOCKER_REPO}"
+TARGET_IMAGE_LATEST="${TARGET_IMAGE}:latest"
+TARGET_IMAGE_VERSIONED="${TARGET_IMAGE}:${TRAVIS_COMMIT}"
+HEROKU_APP_REGISTRY_URL="registry.heroku.com/${HEROKU_APP_NAME}/web"
+
+# Push to Docker Hub
+
+docker login --username $DOCKER_USERNAME --password $DOCKER_PASSWORD
+docker build \
+ --target production \
+ --tag ${TARGET_IMAGE_LATEST} \
+ --tag ${TARGET_IMAGE_VERSIONED} .
+docker push ${TARGET_IMAGE_LATEST}
+docker push ${TARGET_IMAGE_VERSIONED}
+
+# Push to Heroku Registry
+
+docker login --username=_ --password=${HEROKU_AUTH_TOKEN} registry.heroku.com
```

Comment:
Generally it's recommended to pass through the password via stdin rather than directly to make sure it can't end up in a shell's history/log files
E.g. see [the docs](https://docs.docker.com/engine/reference/commandline/login/), example here might be:
`echo $HEROKU_AUTH_TOKEN | docker login --username=_ --password-stdin registry.heroku.com`